internals/overlord/logstate/gatherer_test.go

```diff
@@ -218,6 +219,92 @@ func (s *gathererSuite) TestRetryLoki(c *C) {
 	}
 }
 
+// Test to catch race conditions in gatherer
+func (s *gathererSuite) TestRace(c *C) {
+	target := &plan.LogTarget{
+		Name:     "tgt1",
+		Type:     plan.LokiTarget,
+		Services: []string{"all"},
+		Labels:   map[string]string{"foo": "bar-$SECRET-$SECRET2", "baz": "foo"},
+	}
+
+	g, err := newLogGathererInternal(target, &logGathererOptions{
+		maxBufferedEntries: 2,
+	})
+	c.Assert(err, IsNil)
+
+	svc1 := newTestService("svc1")
+	svc2 := newTestService("svc2")
+	fakeEnv := map[string]string{
+		"SECRET":  "pie",
+		"SECRET2": "pizza",
+	}
+	svc1.config.Environment = fakeEnv
+	svc2.config.Environment = fakeEnv
+
+	buffers := map[string]*servicelog.RingBuffer{
+		svc1.name: svc1.ringBuffer,
+		svc2.name: svc2.ringBuffer,
+	}
+
+	wg := sync.WaitGroup{}
+	doAsync := func(f func()) {
+		wg.Add(1)
+		go func() {
+			f()
+			wg.Done()
+		}()
+	}
+
+	doAsync(func() {
+		g.PlanChanged(&plan.Plan{
+			Services: map[string]*plan.Service{
+				svc1.name: svc1.config,
+			},
+			LogTargets: map[string]*plan.LogTarget{
+				target.Name: target,
+			},
+		}, buffers)
+	})
+
+	doAsync(func() { g.ServiceStarted(svc1.config, svc1.ringBuffer) })
```

Comment:
Can we format these so they're all multi-line to be consistent?

```suggestion
	doAsync(func() {
	    g.ServiceStarted(svc1.config, svc1.ringBuffer)
	})
```